aws/structure.go

```diff
@@ -2138,6 +2139,493 @@ func flattenCognitoIdentityProviders(ips []*cognitoidentity.Provider) []map[stri
+	}
+
+	if v, ok := config["post_confirmation"]; ok {
+		configs.PostConfirmation = aws.String(v.(string))
+	}
+
+	if v, ok := config["pre_authentication"]; ok {
+		configs.PreAuthentication = aws.String(v.(string))
+	}
+
+	if v, ok := config["pre_sign_up"]; ok {
+		configs.PreSignUp = aws.String(v.(string))
+	}
+
+	if v, ok := config["verify_auth_challenge_response"]; ok {
+		configs.VerifyAuthChallengeResponse = aws.String(v.(string))
+	}
+
+	return configs
+}
+
+func flattenCognitoUserPoolLambdaConfig(s *cognitoidentityprovider.LambdaConfigType) []map[string]interface{} {
+	m := map[string]interface{}{}
+
+	if s == nil {
+		return nil
+	}
+
+	if s.CreateAuthChallenge != nil {
+		m["create_auth_challenge"] = *s.CreateAuthChallenge
+	}
+
+	if s.CustomMessage != nil {
+		m["custom_message"] = *s.CustomMessage
+	}
+
+	if s.DefineAuthChallenge != nil {
+		m["define_auth_challenge"] = *s.DefineAuthChallenge
+	}
+
+	if s.PostAuthentication != nil {
+		m["post_authentication"] = *s.PostAuthentication
+	}
+
+	if s.PostConfirmation != nil {
+		m["post_confirmation"] = *s.PostConfirmation
+	}
+
+	if s.PreAuthentication != nil {
+		m["pre_authentication"] = *s.PreAuthentication
+	}
+
+	if s.PreSignUp != nil {
+		m["pre_sign_up"] = *s.PreSignUp
+	}
+
+	if s.VerifyAuthChallengeResponse != nil {
+		m["verify_auth_challenge_response"] = *s.VerifyAuthChallengeResponse
+	}
+
+	return []map[string]interface{}{m}
+}
+
+func expandCognitoUserPoolPasswordPolicy(config map[string]interface{}) *cognitoidentityprovider.PasswordPolicyType {
+	configs := &cognitoidentityprovider.PasswordPolicyType{}
+
+	if v, ok := config["minimum_length"]; ok {
+		configs.MinimumLength = aws.Int64(int64(v.(int)))
+	}
+
+	if v, ok := config["require_lowercase"]; ok {
+		configs.RequireLowercase = aws.Bool(v.(bool))
+	}
+
+	if v, ok := config["require_numbers"]; ok {
+		configs.RequireNumbers = aws.Bool(v.(bool))
+	}
+
+	if v, ok := config["require_symbols"]; ok {
+		configs.RequireSymbols = aws.Bool(v.(bool))
+	}
+
+	if v, ok := config["require_uppercase"]; ok {
+		configs.RequireUppercase = aws.Bool(v.(bool))
+	}
+
+	return configs
+}
+
+func flattenCognitoUserPoolPasswordPolicy(s *cognitoidentityprovider.PasswordPolicyType) []map[string]interface{} {
+	m := map[string]interface{}{}
+
+	if s == nil {
+		return nil
+	}
+
+	if s.MinimumLength != nil {
+		m["minimum_length"] = *s.MinimumLength
+	}
+
+	if s.RequireLowercase != nil {
+		m["require_lowercase"] = *s.RequireLowercase
+	}
+
+	if s.RequireNumbers != nil {
+		m["require_numbers"] = *s.RequireNumbers
+	}
+
+	if s.RequireSymbols != nil {
+		m["require_symbols"] = *s.RequireSymbols
+	}
+
+	if s.RequireUppercase != nil {
+		m["require_uppercase"] = *s.RequireUppercase
+	}
+
+	return []map[string]interface{}{m}
+}
+
+func expandCognitoUserPoolSchema(inputs []interface{}) []*cognitoidentityprovider.SchemaAttributeType {
+	configs := make([]*cognitoidentityprovider.SchemaAttributeType, 0)
```

Comment:
Nitpick, but can't we already allocate fixed-size array here, since we know how many inputs are coming in? i.e. 

```go
configs := make([]*cognitoidentityprovider.SchemaAttributeType, len(inputs), len(inputs))
```
then you'd just use
```go
configs[i] = config
```
instead of append.
But it's really a nitpick, just a matter of memory efficiency.

Reply:
Done! :)